compiler/tests-jsoo/lib-effects-2/dune

```diff
@@ -1,4 +1,13 @@
 (env
+ (with-effects-double-translation
+  (flags
+   (:standard -w -38))
+  (js_of_ocaml
+   ;; separate compilation doesn't yet work when using
+   ;; '--effect=double-translation' since Dune doesn't know it should compile a
+   ;; different version of the dependencies.
+   ;; TODO: remove once support in ocaml/dune#11222 is released.
+   (compilation_mode whole_program)))
```

Comment:
```suggestion
 (with-effects-double-translation
  (flags
   (:standard -w -38)))
```

Reply:
Actually, I think we can just remove the directories `compiler/tests-jsoo/lib-effects-2/` and `compiler/tests-ocaml/effects-2`.